### PR TITLE
Feature/syal 1481

### PR DESCRIPTION
**Pull request by HieuLM-AVISPL**
Hi @MaksimR-AVISPL , Please help me to review and approve the PR of Vaddio EasyIP Mixer. Thanks

### CODE REVIEW

File: src/main/java/com/avispl/symphony/dal/avdevices/mixer/vaddio/easyip/EasyIPMixerCommunicator.java [hunk 1]
```diff
@@ -445,21 +491,21 @@ public void controlProperty(ControllableProperty controllableProperty) throws Ex
...
 						value = checkValidInput(1, 20, value);
 						command = propertyItem.getControlCommand().replace("$1", indexCamera).replace("$2", value);
-						sendCommandToControlDevice(command, value, propertyKey);
+						sendCommandToControlDeviceWithExecutor(command, value, propertyKey);
 						updateCachedDeviceData(cacheKeyAndValue, property, value);
 						break;
 					case CAMERA_PAN:
 						indexCamera = EasyIPMixerMapping.getValueByName(group.replace(EasyIPMixerConstant.VIDEO_INPUT, EasyIPMixerConstant.EMPTY));
-						value = checkValidInput(-150f, 150f, value);
+						value = checkValidInput(-156.3, 151.7, value);
```
Comment (MaksimR-AVISPL) [inline src/main/java/com/avispl/symphony/dal/avdevices/mixer/vaddio/easyip/EasyIPMixerCommunicator.java:499]: These limits are supposed to be this specific?

File: src/main/java/com/avispl/symphony/dal/avdevices/mixer/vaddio/easyip/EasyIPMixerCommunicator.java [hunk 2]
```diff
@@ -445,21 +491,21 @@ public void controlProperty(ControllableProperty controllableProperty) throws Ex
...
 						command = propertyItem.getControlCommand().replace("$1", indexCamera).replace("$2", value);
-						sendCommandToControlDevice(command, value, propertyKey);
+						sendCommandToControlDeviceWithExecutor(command, value, propertyKey);
 						updateCachedDeviceData(cacheKeyAndValue, property, value);
 						break;
 					case CAMERA_TILT:
 						indexCamera = EasyIPMixerMapping.getValueByName(group.replace(EasyIPMixerConstant.VIDEO_INPUT, EasyIPMixerConstant.EMPTY));
-						value = checkValidInput(-30, 90, value);
+						value = checkValidInput(-30, 92.5, value);
```
Comment (MaksimR-AVISPL) [inline src/main/java/com/avispl/symphony/dal/avdevices/mixer/vaddio/easyip/EasyIPMixerCommunicator.java:506]: Also these, is there any chance they will change in future or for other devices?

File: src/main/java/com/avispl/symphony/dal/avdevices/mixer/vaddio/easyip/EasyIPMixerCommunicator.java [hunk 3]
```diff
@@ -1106,14 +1289,20 @@ private void populateCrossPointGain(Map<String, String> stats, List<AdvancedCont
...
 			}
+			if (response.contains("Invalid routing request")) {
+				throw new IllegalArgumentException(String.format("Error when control %s, Route is unavailable", name));
+			}
+			if (response.contains(EasyIPMixerConstant.ERROR_RESPONSE) || !response.contains(EasyIPMixerConstant.OK) || response.contains(EasyIPMixerConstant.ERROR)) {
+				throw new IllegalArgumentException(String.format("Error when control %s, Syntax error command: %s", name, response));
+			}
 		} catch (Exception e) {
-			throw new IllegalArgumentException(String.format("Can't control %s with %s value.", name, value), e);
+			throw new IllegalArgumentException(String.format("Can't control %s with %s value. ", name, value) + e.getMessage());
```
Comment (MaksimR-AVISPL) [inline src/main/java/com/avispl/symphony/dal/avdevices/mixer/vaddio/easyip/EasyIPMixerCommunicator.java]: please consider using e.getMessage() either as String.format param or IllegalArgumentException constructor parameter instead

**Review by MaksimR-AVISPL — APPROVED** [review]

File: src/main/java/com/avispl/symphony/dal/avdevices/mixer/vaddio/easyip/EasyIPMixerCommunicator.java (hunk 1 above)
Comment (HieuLM-AVISPL) [reply src/main/java/com/avispl/symphony/dal/avdevices/mixer/vaddio/easyip/EasyIPMixerCommunicator.java:499]: > These limits are supposed to be this specific?

image
There is a difference between real device and documentation.
According to the discussion, Nicolas chose the range according to real device

**Pull request merged**